consensus/istanbul/backend/handler.go

```diff
@@ -266,10 +267,28 @@ func (sb *Backend) UpdateMetricsForParentOfBlock(child *types.Block) {
 	inParentSeal := childExtra.ParentAggregatedSeal.Bitmap.Bit(gpValSetIndex) != 0
 	if inParentSeal {
 		sb.blocksElectedAndSignedMeter.Mark(1)
+		sb.blocksElectedButNotSignedGauge.Update(0)
 	} else {
 		sb.blocksElectedButNotSignedMeter.Mark(1)
+		sb.blocksElectedButNotSignedGauge.Update(sb.blocksElectedButNotSignedGauge.Value() + 1)
 		sb.logger.Warn("Elected but didn't sign block", "number", number-1, "address", sb.ValidatorAddress())
 	}
+
+	// Report downtime events
+	if sb.blocksElectedButNotSignedGauge.Value() >= 12 {
+		sb.blocksDowntimeEventMeter.Mark(1)
+		sb.logger.Error("Elected but haven't signed >= 12 blocks in a row", "missed block count", sb.blocksElectedButNotSignedGauge.Value(), "number", number-1, "address", sb.ValidatorAddress())
+	}
+
+	// Clear downtime counter on end of epoch.
+	if istanbul.IsLastBlockOfEpoch(number-1, sb.config.Epoch) {
+		sb.blocksElectedButNotSignedGauge.Update(0)
+	}
+
+	// missed round as proposer
+	if missedRounds > 0 && parentHeader.Coinbase == sb.ValidatorAddress() {
+		sb.blocksMissedRoundsAsProposerMeter.Mark(missedRounds)
```

Comment:
Is this metric an accumulator for the number of missed rounds when this validator is the proposer?

IIUC, right now it's the total number of missed rounds (regardless of which validator is the proposer) for sequences right after this validator proposed a finalized block.

Reply:
It's supposed to be the number of missed round when the validator is the proposer.

My understanding is that missedRounds while coming from the child block, refers back to the parent block, and the parent coinbase is the proposer for that block.

Reply:
Ah, just now seeing that the `missedRound` is from the parent aggregated seal.

Also, the proposer should be based on round, not block number.  E.g. for a specific block number, the proposer will be rotated as rounds increase for that block: https://github.com/celo-org/celo-blockchain/blob/master/consensus/istanbul/validator/selectors.go#L54

Reply:
Ok, might need a rename, something like `potentialMissedProposalOpportunity`, but this still remains tricky with how proposer rotations are done.